BeerOverflowWindowsApp/BarProviders/GetBarListFacebook.cs

```diff
@@ -53,20 +52,21 @@ private IEnumerable<Place> GetBarData(string latitude, string longitude, string
         {
             RegexTools.LocationDataTextIsCorrect(latitude, longitude, radius);
             var link = string.Format(_apiLink, _accessToken, latitude, longitude, radius, _requestedFields, _category);
-            var jsonStream = await _fetcher.GetHttpStreamAsync(link);
-            var barList = JsonConvert.DeserializeObject<PlacesResponse>(jsonStream).data;
+            var deserialized = await FetcherAndDeserializer.FetchAndDeserializeAsync<PlacesResponse>(link, _fetcher);
+            var barList = deserialized.data;
             return barList;
         }
 
         private List<BarData> PlaceListToBarList(IEnumerable<Place> resultData)
         {
+            var allowedCategories = string.Join(",", _allowedCategories.Split(',').ToList().SelectMany(category => category.Split('|')).Where((c, i) => i % 2 == 0));
```

Comment:
Maybe its just my lack of knowledge, but in Where((c, i) => i % 2 == 0), whats the use of parameter c?

Reply:
This one collects elements, if the index of element is even (index % 2 == 0). Not covered in our course. Sadly, it's googled code.